File: src/main/java/com/vspk/bookrest/service/UserAuthService.java
package com.vspk.bookrest.service;

import com.vspk.bookrest.dto.AuthUserDetailsDto;
import com.vspk.bookrest.dto.RegisterUserDetailsDto;
import org.springframework.http.ResponseEntity;

public interface UserAuthService {
    ResponseEntity<?> authenticate(AuthUserDetailsDto authUserDetailsDto);
    ResponseEntity<?> register(RegisterUserDetailsDto registerUserDetailsDto);
}

File: src/main/java/com/vspk/bookrest/service/UserAuthService.kt
package com.vspk.bookrest.service

import com.vspk.bookrest.dto.AuthUserDetailsDto
import com.vspk.bookrest.dto.RegisterUserDetailsDto
import com.vspk.bookrest.payload.LoginResponse
import com.vspk.bookrest.payload.RegistrationResponse
import org.springframework.http.ResponseEntity

interface UserAuthService {
    fun authenticate(authUserDetailsDto: AuthUserDetailsDto): ResponseEntity<LoginResponse>
    fun register(registerUserDetailsDto: RegisterUserDetailsDto): ResponseEntity<RegistrationResponse>
}